src/types/modifications/ReserveRequirement.jl

```diff
@@ -6,178 +6,440 @@ Representation of reserve requirement, such that the sum of eligible power injec
+
+
+
+
+    # Create column for credit level, set to zeros to start
+    credits = Container[ByNothing(0.0) for i in axes(gen,1)]
     add_table_col!(data, :gen, mod.name, credits, CreditsPerMWCapacity,
         "Credit level for reserve requirement $(mod.name).  This gets multiplied by the power capacity in the reserve requirement constraint.")
 
-    for gen_idx in gen_idxs
+    for gen_idx in axes(gen, 1)
         credits[gen_idx] = Container(get_credit(c_gen, data, gen[gen_idx, :]))
     end
 
-    # Set up the capacity constraint
-    cons_name = Symbol("cons_pcap_$(mod.name)")
+    # Pull out some of the variables/expressions
     pcap_gen = model[:pcap_gen]::Matrix{VariableRef}
     plserv_bus = model[:plserv_bus]::Array{AffExpr, 3}
-    expr_pcap = @expression(
+
+    # Create expression for total reserve power by subarea
+    pres_subarea = @expression(
         model,
-        [yr_idx in 1:nyr, hr_idx in 1:nhr],
-        sum(
-            pcap_gen[gen_idx, yr_idx] * credits[gen_idx][yr_idx, hr_idx] 
-            for gen_idx in gen_idxs
-        )
+        [sa_idx in axes(subareas,1), yr_idx in 1:nyr, hr_idx in 1:nhr],
+        AffExpr(0.0)
     )
-    
+
+    # Create expression for reserve power from generators by subarea
+    pres_gen_subarea = @expression(
+        model,
+        [sa_idx in axes(subareas,1), yr_idx in 1:nyr, hr_idx in 1:nhr],
+        AffExpr(0.0)
+    )
+
+    for sa_idx in axes(subareas,1)
+        gen_idxs = gen_idx_sets[sa_idx]
+        for gen_idx in gen_idxs, yr_idx in 1:nyr, hr_idx in 1:nhr
+            add_to_expression!(pres_gen_subarea[sa_idx, yr_idx, hr_idx], pcap_gen[gen_idx, yr_idx], credits[gen_idx][yr_idx, hr_idx])
+        end
+    end
+
+    # Add the power reserves from generators to the total expression.
+    for sa_idx in axes(subareas,1), yr_idx in 1:nyr, hr_idx in 1:nhr
+        add_to_expression!(pres_subarea[sa_idx, yr_idx, hr_idx], pres_gen_subarea[sa_idx, yr_idx, hr_idx])
+    end
+
+
+    # Get requirement by bus and year
+    bus_yr_idx_2_req = fill(-1.0, (nbus, nyr))
+    years_sym = Symbol.(years)
+    for req_idx in axes(requirements,1)
+        for bus_idx in requirements.bus_idx_sets[req_idx]
+            for (yr_idx, year) in enumerate(years_sym)
+                bus_yr_idx_2_req[bus_idx, yr_idx] = get(requirements[req_idx,:], year, -1.0)
+            end
+        end
+    end
+
+    # Set up expression for the reserve requirement pres_req_subarea
     if mod.load_type == "plnom"
-        expr_pres = @expression(
+        pres_req = @expression(
             model,
-            [yr_idx in 1:nyr, hr_idx in 1:nhr],
-            (1 + get(requirements, Symbol(years[yr_idx]), -1.0)) * 
-            sum(
-                get_plnom(data, bus_idx, yr_idx, hr_idx)
-                for bus_idx in bus_idxs
-            )
+            [bus_idx in axes(bus, 1), yr_idx in 1:nyr, hr_idx in 1:nhr],
+            (1 + bus_yr_idx_2_req[bus_idx, yr_idx]) * get_plnom(data, bus_idx, yr_idx, hr_idx)
```

Comment:
so is this now setting a requirement at each bus instead of over the sum of the buses in the area?

Reply:
Not quite - this is simply indicating how much of the area-wide requirement comes from each bus.  It's helpful for results aggregation/attributing welfare.